contract/models/account_analytic_account.py

```diff
@@ -75,15 +85,60 @@ def _onchange_contract_template_id(self):
             )):
                 self[field_name] = self.contract_template_id[field_name]
 
-    @api.onchange('recurring_invoices')
-    def _onchange_recurring_invoices(self):
-        if self.date_start and self.recurring_invoices:
+    @api.onchange('date_start')
+    def _onchange_date_start(self):
```

Comment:
Question, what is the reasoning for not testing on recurring_invoices?

Reply:
With this, now you don't need to care about that field: the recurring_next_invoice will always be synchronized.

Reply:
It looks like we're also implicitly assuming that it's a recurring contract in the event of a `start_date` (IMO a great simplification). With that though, could we then just outright remove the `recurring_next_invoice` field?

Reply:
Sorry `recurring_invoices` field, not `recurring_next_invoice`

Reply:
Previous code assumes the same: it took `date_start` when you mark `recurring_invoices` field. What you are talking about is to take `today` when marking `recurring_invoices`. It can be another thing to add indeed.

What I don't see correct is to remove `recurring_invoices` field, as it serves a lot for UI usability and user experience. Having that check, you clean the interface out of fields that aren't important if you don't use recurring invoicing. Note that in this PR, I have improved also the experience automarking by default the field when clicking on create when you are in the "Contracts" menu.

Reply:
Thanks for the elaboration Pedro. Does it make sense to actually store `recurring_invoices`, given that it is really just a functional field for triggering of specific workflows?

Just spitballing other upgrades here seeing as we have these nice improvements in similar areas. Not blocking

Reply:
Well, I prefer to keep it right now, or we would need to change a lot of filters, context values...